Sources/MongoSwift/ConnectionString.swift

```diff
@@ -174,7 +174,28 @@ internal class ConnectionString {
         guard let optsDoc = mongoc_uri_get_options(self._uri) else {
             return nil
         }
-        return BSONDocument(copying: optsDoc)
+        var copy = BSONDocument(copying: optsDoc)
+
+        if let authSource = self.authSource {
```

Comment:
libmongoc doesn't put a lot of these into the returned document so we have to add them manually to get the full list

Reply:
Is this a bug in libmongoc then?

Reply:
they document the method as only returning the options provided after the ? of a URI. 

all of the auth info in a URI comes before that. 

however, that still leaves a few cases, all of which they appear to be aware of as they work around them in their own version of these tests:
- tag sets and compressors: https://github.com/mongodb/mongo-c-driver/blob/c64955f98d7ddb6ab25e14501ecd7d175acc4ab1/src/libmongoc/tests/test-mongoc-connection-uri.c#L127
- read concern level: https://github.com/mongodb/mongo-c-driver/blob/c64955f98d7ddb6ab25e14501ecd7d175acc4ab1/src/libmongoc/tests/test-mongoc-connection-uri.c#L227